### DISCUSSION TITLE
getting empty string using the STT from Twillio

### DISCUSSION THREAD

**Discussion by KADURKA-ANIL** [deepgram/community#833] [General help]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

i have tried with the twilio proxy stereo and proxy mono i got the empty stirng there ryt over there also gettng the same empty string here .please do check the reason that can be done 

import asyncio
import json
import base64
import websockets
import ssl
import wave
from deepgram import DeepgramClient, DeepgramClientOptions, LiveTranscriptionEvents, LiveOptions

API_KEY = ''

# Configure the DeepgramClientOptions to enable KeepAlive for maintaining the WebSocket connection
config = DeepgramClientOptions(options={"keepalive": "true"})

# Create a Deepgram client
deepgram = DeepgramClient(API_KEY, config)

# Create the websocket connection
dg_connection = deepgram.listen.asynclive.v("1")

async def on_open(self, open, **kwargs):
    print('Connection Open')
    print(f"{open}")

async def on_message(self, result, **kwargs):
    print('Received message')
    sentence = result.channel.alternatives[0].transcript
    if len(sentence) == 0:
        print("len=0")
        return
    print(f"Speaker: {sentence}")

async def on_metadata(self, metadata, **kwargs):
    print('Received metadata')
    print(f"{metadata}")

async def on_speech_started(self, speech_started, **kwargs):
    print('Speech started')
    print(f"{speech_started}")

async def on_utterance_end(self, utterance_end, **kwargs):
    print('Utterance end')
    print(f"{utterance_end}")

async def on_close(self, close, **kwargs):
    print('Connection Closed')
    print(f"{close}")

async def on_error(self, error, **kwargs):
    print('Error encountered')
    print(f"{error}")

async def on_unhandled(self, unhandled, **kwargs):
    print('Unhandled message')
    print(f"{unhandled}")

# Set up event handlers for Deepgram WebSocket
dg_connection.on(LiveTranscriptionEvents.Open, on_open)
dg_connection.on(LiveTranscriptionEvents.Transcript, on_message)
dg_connection.on(LiveTranscriptionEvents.Metadata, on_metadata)
dg_connection.on(LiveTranscriptionEvents.SpeechStarted, on_speech_started)
dg_connection.on(LiveTranscriptionEvents.UtteranceEnd, on_utterance_end)
dg_connection.on(LiveTranscriptionEvents.Close, on_close)
dg_connection.on(LiveTranscriptionEvents.Error, on_error)
dg_connection.on(LiveTranscriptionEvents.Unhandled, on_unhandled)

# Set up transcription options
options = LiveOptions(
    model="nova-2",
    language="en-US",
    smart_format=True,
    encoding="linear16",
    channels=1,
    sample_rate=16000,
    interim_results=True,
    utterance_end_ms=1000,
    vad_events=True,
    endpointing=300,
)

async def handle_media(socket):
    print("Starting media handler")
    connected = await dg_connection.start(options)
    if not connected:
        print("Failed to connect to Deepgram")
        return
    print("Successfully connected to Deepgram")

    audio_buffer  = bytearray(b'')

    try:
        async for message in socket:
            data = json.loads(message)
            if data['event'] == 'media':
                media = data['media']
                payload = media['payload']
                chunk = base64.b64decode(payload)

                audio_buffer.extend(chunk)

            else:
                pass
# Parameters for the WAV file
        num_channels = 1  # Mono
        sample_width = 2  # 2 bytes per sample
        frame_rate = 44100  # 44.1 kHz
        num_frames = 0  # We will add frames later

        # Open a WAV file for writing
        with wave.open('output.wav', 'wb') as wav_file:
            # Set the parameters
            wav_file.setnchannels(num_channels)
            wav_file.setsampwidth(sample_width)
            wav_file.setframerate(frame_rate)
            
            # Write the audio data
            #audio_data = b'\x00\x00' * 44100  # 1 second of silence
            wav_file.writeframes(audio_buffer)

            print("WAV file has been created.")

                #await dg_connection.send(chunk)
            
    finally:
        pass
        #with open('/Users/gautamgurrala/websocket/out.wav', 'wb') as wavfile:

        #await dg_connection.finish()

async def test(websocket):
    async for message in websocket:
        print(message)
        await websocket.send(message)

async def echo(websocket, path):
    if path == '/media':
        print('Received /media path')
        await handle_media(websocket)
    elif path == '/test':
        await test(websocket)

if __name__ == '__main__':
    ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
    ssl_context.load_cert_chain('cert.pem', 'key.pem')

    start_server = websockets.serve(echo, "localhost", 8765, ssl=ssl_context)

    asyncio.get_event_loop().run_until_complete(start_server)
    asyncio.get_event_loop().run_forever()

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.
